routing/router.go

```diff
@@ -2846,10 +2846,17 @@ func (r *ChannelRouter) BuildRoute(amt *lnwire.MilliSatoshi,
 	var pathEdges []*channeldb.CachedEdgePolicy
 	receiverAmt := runningAmt
 	for i, unifier := range unifiers {
+		var fromNode route.Vertex
```

Comment:
Should we move this into the `if edge == nil {` branch to make it more explicit we're only using this in the error/not found case?
Also:
```
fromNode := source
if i > 0 {
    fromNode := hopts[i-1]
}
```
?